lib/pages/home_page.dart

```diff
@@ -45,22 +46,29 @@ class _HomePageState extends State<HomePage> {
   Future<void> getChallengeIfAlreadyStarted() async {
     final challenge = await Firestore.instance.getChallengePending();
     if (challenge != null) {
+      pos = await Geolocation.instance.position;
       setState(() {
         chalLat = challenge['lat'];
         chalLng = challenge['lng'];
         daily = Marker(
-            point: LatLng(challenge['lat'], challenge['lng']),
+            point: LatLng(chalLat, chalLng),
             builder: (context) =>
-                const Icon(Icons.room, color: Colors.red, size: 50));
+                const Icon(Icons.room, color: Colors.red, size: 25));
         isAlreadyStarted = true;
+        
+
         _timer = Timer.periodic(const Duration(seconds: 10), (_) async {
           Position pos = await Geolocation.instance.position;
-          if (DistCalculator.instance.checkDist(
+          int temp = DistCalculator.instance.getDist(
               Location(pos.latitude, pos.longitude),
-              Location(challenge['lat'], challenge['lng']))) {
-            finishChallenge();
+              Location(challenge['lat'], challenge['lng']));
+          if (temp <= 50) {
+            await finishChallenge();
           }
```

Comment:
Vzhledem k tomu, že sám nevíš co je `temp`... doporučuju pojmenovat tak aby bylo vidět co to dělá 😄